code/__HELPERS/_lists.dm

```diff
@@ -205,6 +205,22 @@
 
 	return null
 
+/proc/pickweightAllowZero(list/L) //The original pickweight proc will sometimes pick entries with zero weight.  I'm not sure if changing the original will break anything, so I left it be.

```

Comment:
@MrPerson is rewriting rng and I'm sure they have a better solution than this

Reply:
#30087 Does not touch the pickweight proc at all.  I'll make a note in that PR suggesting he does so.